### ISSUE TITLE
CELL/FULL not always forwarded

### ISSUE THREAD

**Issue by cmeeren** [matortheeternal/smash#204]
I'm new to Mator Smash and Skyrim modding in general, so I'm sorry if I'm using the wrong terms.

I'm trying to create a Smashed patch for my load order. I'm using `MLU.esp` from [OMEGA](https://www.nexusmods.com/skyrimspecialedition/mods/17653), as well as `ELE_SSE.esp` (Enhanced Lightning for ENB). The former edits the name of a lot of interior cells (`CELL/FULL`), whereas the latter edits the lightning in most interior cells. I want Mator Smash to forward(?) the name changes from the OMEGA `MLU.esp` and the lighting changes from ELE.

I have `MLU.esp` set to `Smash.ForceAll` as per the recommendations on Mator Smash's Nexus page. In order to make Mator.Smash use the name changes in MLU.esp, I have created a new Smash setting for all mods that load below MLU.esp. This setting is a copy of `Smash.All`, just with `CELL/FULL` unchecked.

This works for most cell records, but there is a handful that do not seem to be picked up and included in the Smashed patch.

Do you have any idea what (if anything) could be wrong, or how I can ensure that the `CELL/FULL` changes from `MLU.esp` are always included (without creating a manual patch, which is my plan B)?

Below are screenshots of 1) a record where CELL/FULL was included in the Smashed patch, and 2) a record where it was not.

![included](https://user-images.githubusercontent.com/7766733/42188500-6d3d3068-7e54-11e8-9bd1-ca5fdfabc188.png)

![not included](https://user-images.githubusercontent.com/7766733/42188501-6d5f4bb2-7e54-11e8-8600-a893c9571892.png)

**Comment by Mincol** [comment]
This might be related to https://github.com/matortheeternal/smash/issues/203.

For ELE changes to be propagated, you can use my settings called BashC.Light-My and can be downloaded  https://github.com/matortheeternal/smash/files/2146516/Skyrim.zip Extract file called BashC.Light-My, store it in Mator.Smash\settings\Skyrim and mark ELE to use BashC.Light-My instead of BashC.Light

**Comment by Mincol** [comment]
Please enlighten my why Smash.force is needed, when you do not have conflicts in the names? Thank you! I would like to learn new stuff.

> Smash.ForceAll is a special smash setting which forces smash to use records from the plugin it is applied to, and ignore conflicts with these records unless they belong to a plugin which requires the plugin as a master.  This should be useful to users running Requiem.

**Comment by cmeeren** [comment]
Thanks for the Bash settings. But why isn't the lighting carried over? I even tried a custom setting where I forced the relevant sub-records from ELE, but Smash still didn't use the lighting changes.

Regarding Force, I use it for MLU since it's officially recommended.

Also, yes, it might be related to #203.

**Comment by cmeeren** [comment]
@Mincol no matter what I do and which settings I use, I can't get the ELE changes propagated.

**Comment by Mincol** [comment]
@cmeeren make sure you do not use Smash.ForcaAll, but Smash.All. I will try it tonight.

**Comment by Mincol** [comment]
I can reproduce it:
![image](https://user-images.githubusercontent.com/7746255/42239974-0006b99e-7f06-11e8-83a0-f9022cb6433b.png)
Cell \ Block 0 \ Sub-Block 0 is **missing** from smashed patch.

**Comment by cmeeren** [comment]
Note that most of the missing records are included when the "Preserve ITPO" setting in Mator Smash is checked. However, the ELE lighting changes are still not forwarded.

**Comment by matortheeternal** [comment]
Question: If you disable "Remove ITPOs", is the record in the smashed patch with a FULL name identical to what is in ELE_SSE.esp?

**Comment by Mincol** [comment]
With "Preserve ITPOs"
![image](https://user-images.githubusercontent.com/7746255/42615664-76bb0058-85ab-11e8-9612-f859049efde9.png)
What language is used to write Smash? Is it Pascal? Is tooling free? I did Pascal at school more than 15 years ago and do java for money now...

**Comment by cmeeren** [comment]
@matortheeternal I'm on holiday and can't test at the moment, but IIRC:

* Without Preserve ITPOs: Behavior as originally reported
* With Preserve ITPOs: Behavior as described in my previous comment - more smashed records have correct FULL, but lighting from ELE is not included.

**Comment by matortheeternal** [comment]
@Mincol looks like the issue is detecting ITPO status then.  The problem isn't that the FULL isn't getting forwarded, it's that Smash removes the record as an ITPO when it isn't.

Smash is Delphi (object oriented pascal) because the xEdit framework (which is what it's built on) is Delphi.  You can download Delphi Start Edition from embarcadero for free.

@cmeeren The lighting not being included is a separate issue if that's happening.

**Comment by cmeeren** [comment]
Yes, I'm 98% sure the lighting was not included, at least for a large number of records (I only looked at the conflicts, so there might have been records where things worked, but there sure was a lot of faulty ones). A temporary workaround I did was to create a custom CR patch where I copied the ELE cell records and forward the FULL change from the OMEGA `MLU.esp`.

**Comment by matortheeternal** [comment]
If you could make a separate github issue about lighting not forwarding with screenshots I'll look into fixing it for the next release.

**Comment by jgoulet** [comment]
Posted something on Nexus but just came across this thread.  I'm having a similar issue with some textures:

After smashing my plugins, some of the NPCs didn't have any hand textures.  Turns out, they all used FarmerGloves03AA, which was modified in my patch (between Remodeled Armor for CBBE, USSEP, and Skyrim.esm).  When I pulled up the comparison pane, for some reason the First Person Flags for the Smashed Patch were blank, even though each of the other files had them as 33 - Hands.  USSEP had 36 - Ring additionally, but I'm not sure that would cause this behavior because I assumed the Remodeled Armor mod would just take precedence.  Once I added the 33 - Hands flag to the patch, voila, the hands appear in game!  Additionally, for some reason they were flagged as Light Armor in the patch even though they are flagged as Clothing in all the others (I'm thinking that may be the default if you add a new flag though).

For good measure, I went down the list and found FineBoots02AA in the same state.  All the flags were identical for the 3 files, but somehow didn't make it to the patch.

Does this sound related to the Preserve ITPO setting?

Edit: that setting wasn't enabled.  So it sounds like I have the same issue as OP's lighting from ELE not propagating.  Was a fix ever determined for this?
